modules/hivemq/src/main/java/org/testcontainers/hivemq/HiveMQContainer.java

```diff
@@ -0,0 +1,651 @@
+                    }
+                });
+            }
+        });
+    }
+
+    @Override
+    public void start() {
+        super.start();
+        if (controlCenterEnabled) {
+            logger.info("The HiveMQ Control Center is reachable under: http://localhost:{}", getMappedPort(CONTROL_CENTER_PORT));
+        }
+    }
+
+    /**
+     * Adds a wait condition for the extension with this name.
+     * <p>
+     * Must be called before the container is started.
+     *
+     * @param extensionName the extension to wait for
+     * @return self
+     */
+    public @NotNull HiveMQContainer waitForExtension(final @NotNull String extensionName) {
+        final String regEX = "(.*)Extension \"" + extensionName + "\" version (.*) started successfully(.*)";
+        waitStrategy.withRegEx(regEX);
+        return self();
+    }
+
+    /**
+     * Adds a wait condition for this {@link HiveMQExtension}
+     * <p>
+     * Must be called before the container is started.
+     *
+     * @param extension the extension to wait for
+     * @return self
+     */
+    public @NotNull HiveMQContainer waitForExtension(final @NotNull HiveMQExtension extension) {
+        return this.waitForExtension(extension.getName());
+    }
+
+    /**
+     * Enables the possibility for remote debugging clients to connect.
+     * <p>
+     * Must be called before the container is started.
+     *
+     * @return self
+     */
+    public @NotNull HiveMQContainer withDebugging() {
+        addExposedPorts(DEBUGGING_PORT);
+        withEnv("JAVA_OPTS", "-agentlib:jdwp=transport=dt_socket,address=0.0.0.0:" + DEBUGGING_PORT + ",server=y,suspend=y");
+        return self();
+    }
+
+    /**
+     * Sets the logging {@link Level} inside the container.
+     * <p>
+     * Must be called before the container is started.
+     *
+     * @param level the {@link Level}
+     * @return self
+     */
+    public @NotNull HiveMQContainer withLogLevel(final @NotNull Level level) {
+        this.withEnv("HIVEMQ_LOG_LEVEL", level.name());
+        return self();
+    }
+
+    /**
+     * Wraps the given class and all its subclasses into an extension
+     * and puts it into '/opt/hivemq/extensions/{extension-id}' inside the container.
+     * <p>
+     * Must be called before the container is started.
+     *
+     * @param hiveMQExtension the {@link HiveMQExtension} of the extension
+     * @return self
+     */
+    public @NotNull HiveMQContainer withExtension(final @NotNull HiveMQExtension hiveMQExtension) {
+        try {
+            final File extension = createExtension(hiveMQExtension);
+            final MountableFile mountableExtension = MountableFile.forHostPath(extension.getPath(), MODE);
+            withCopyFileToContainer(mountableExtension, "/opt/hivemq/extensions/" + hiveMQExtension.getId());
+        } catch (final Exception e) {
+            throw new ContainerLaunchException(e.getMessage() == null ? "" : e.getMessage(), e);
```

Comment:
Can you explain the reason @YannickWeber?

Reply:
The `hiveMQExtension.createExtension(hiveMQExtension)` throws a checked Exception. I intercept it here because I do not want to have a checked Exception in the fluent API.

Reply:
Gotcha, then this makes sense.